### ISSUE TITLE
[Bug]: rich notification not working on iOS

### ISSUE THREAD

**Issue by huextrat** [braze-inc/braze-react-native-sdk#234]
### Which Platforms?

iOS

### Which React Native Version?

0.72.6

### Which @braze/react-native-sdk SDK version?

7.0.0

### Repro Rate

100%

### Steps To Reproduce

Base implementation is working as expected to receive push notifications.

I tried to add rich notifications to display an image in a notification, but I couldn't get it to work properly. I followed this documentation https://braze-inc.github.io/braze-swift-sdk/tutorials/braze/b2-rich-push-notifications

Previously, on `react-native-appboy-sdk` we had a notification service which works correctly, but I'm not able to make it work with this new SDK.

Here is my config in `AppDelegate.mm`

```obj-c
- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  ....
  BRZConfiguration *configuration = [[BRZConfiguration alloc] initWithApiKey:brazeApiKey endpoint:brazeEndpoint];
  configuration.triggerMinimumTimeInterval = 1;
  configuration.logger.level = BRZLoggerLevelInfo;
  Braze *braze = [BrazeReactBridge initBraze:configuration];
  AppDelegate.braze = braze;
  [self registerForPushNotifications];
  [[BrazeReactUtils sharedInstance] populateInitialUrlFromLaunchOptions:launchOptions];
  return [super application:application didFinishLaunchingWithOptions:launchOptions];
}



- (void)registerForPushNotifications {
  [UIApplication.sharedApplication registerForRemoteNotifications];
  UNUserNotificationCenter *center = UNUserNotificationCenter.currentNotificationCenter;
  [center setNotificationCategories:BRZNotifications.categories];
  center.delegate = self;
  UNAuthorizationOptions options = UNAuthorizationOptionAlert | UNAuthorizationOptionSound | UNAuthorizationOptionBadge;
  if (@available(iOS 12.0, *)) {
    options = options | UNAuthorizationOptionProvisional;
  }
  [center requestAuthorizationWithOptions:options
                        completionHandler:^(BOOL granted, NSError *_Nullable error) {
                          NSLog(@"Notification authorization, granted: %d, "
                                @"error: %@)",
                                granted, error);
  }];
}
```

And my `NotificationService.swift` :

```swift
import BrazeNotificationService

class NotificationService: BrazeNotificationService.NotificationService {}
```

### Expected Behavior

Adding rich notification should work with RN and `BrazeNotificationService` or a custom implementation in NotificationService.

### Actual Incorrect Behavior

Rich notification are not working (i.e: images are not displayed)

### Additional Information

Is there any sample code with RN SDK and iOS rich notification implementation ? I've only relied on the native iOS documentation, since there's nothing specified for the case of iOS rich notifications with the RN SDK.

**Comment by huextrat** [comment]
Thanks for your answer @jerielng.

As indicated in the description of the issue, this is what I followed, but I still have the same problem, the notification is well received, but the media is not displayed.

Using the old SDK https://www.npmjs.com/package/react-native-appboy-sdk everything is working well.

**Comment by hokstuff** [comment]
@bheftyes If you import the npm package [@braze/react-native-sdk](https://www.npmjs.com/package/@braze/react-native-sdk), that will [already import `braze-swift-sdk` automatically](https://github.com/braze-inc/braze-react-native-sdk/blob/0e2c43220cd9d7c593cf40ab0b1305177ba62edb/braze-react-native-sdk.podspec#L23-L25).

Rich Push requires importing the [BrazeNotificationService](https://github.com/braze-inc/braze-swift-sdk/#:~:text=%E2%9C%85-,BrazeNotificationService,-Notification%20service%20extension) which would be a separate target from your main application. Since you are using the Braze React Native SDK, you'd need to import this as an additional part of your Podfile ([like shown here](https://github.com/braze-inc/braze-swift-sdk/blob/d3963492b9ae0dd2c28a41201707b1723a7d7c94/Examples/Swift/Podfile#L93), but added to [your React Native app's Podfile](https://github.com/braze-inc/braze-react-native-sdk/blob/master/BrazeProject/ios/Podfile)).

**Comment by jerielng** [comment]
@huextrat By the way, we're looking to get some samples integrated into our React Native sample app, so you will be able to pattern match off of that. In the meantime, if you are still facing issues with your integration, here are some troubleshooting steps you can try:
* Does your notification service extension target matches the minimum deployment of your main target?
* Are the relevant fields added to your Info.plist, particularly the `NSExtensionPrincipalClass`?
* Is the `BrazeNotificationService` framework added to the Frameworks and Libraries list under your General settings?
* Likewise, is your app extension added to the Frameworks, Libraries, and Embedded Content under your main target?

**Comment by huextrat** [comment]
Hi @jerielng, I've a pretty straightforward implementation which is looking like this:

- Minimum deployment target is the same
- `NSExtensionPrincipalClass` is equals to `NotificationService` which is the name of my extension
- `BrazeNotificationService` is well added as a Pod on `NotificationService` target + added in the `NotificationService` target in `Frameworks and Librairies` section
- `NotificationService.appex` is present, the main target under `Frameworks and Librairies` section

Notification are well received, but the media (image) is not displayed. With previous implementation of `react-native-appboy-sdk` it was working well, and I send the notification through Braze dashboard.

If you have any idea, would be glad to hear 👍 

AppDelegate.mm
```obj-c
- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  ...
  BRZConfiguration *configuration = [[BRZConfiguration alloc] initWithApiKey:brazeApiKey endpoint:brazeEndpoint];
  configuration.triggerMinimumTimeInterval = 1;
  configuration.logger.level = BRZLoggerLevelInfo;
  configuration.push.automation = [[BRZConfigurationPushAutomation alloc] initEnablingAllAutomations:YES];
  
  Braze *braze = [BrazeReactBridge initBraze:configuration];
  AppDelegate.braze = braze;

  [self registerForPushNotifications];
  [[BrazeReactUtils sharedInstance] populateInitialUrlFromLaunchOptions:launchOptions];
  ...
}

- (void)registerForPushNotifications {
  UNUserNotificationCenter *center = UNUserNotificationCenter.currentNotificationCenter;
  [center setNotificationCategories:BRZNotifications.categories];
  center.delegate = self;
  [UIApplication.sharedApplication registerForRemoteNotifications];
}

- (void) application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo fetchCompletionHandler:(void (^)(UIBackgroundFetchResult))completionHandler {
  BOOL processedByBraze = AppDelegate.braze != nil && [AppDelegate.braze.notifications handleBackgroundNotificationWithUserInfo:userInfo
                                                                                                           fetchCompletionHandler:completionHandler];
    if (processedByBraze) {
      return;
    }

    completionHandler(UIBackgroundFetchResultNoData);
}

- (void)userNotificationCenter:(UNUserNotificationCenter *)center didReceiveNotificationResponse:(UNNotificationResponse *)response withCompletionHandler:(void (^)(void))completionHandler {
  [[BrazeReactUtils sharedInstance] populateInitialUrlForCategories:response.notification.request.content.userInfo];
    BOOL processedByBraze = AppDelegate.braze != nil && [AppDelegate.braze.notifications handleUserNotificationWithResponse:response
                                                                                                      withCompletionHandler:completionHandler];
    if (processedByBraze) {
      return;
    }

    completionHandler();
}

- (void)application:(UIApplication*)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData*)deviceToken
{
  [AppDelegate.braze.notifications registerDeviceToken:deviceToken];
}

static Braze *_braze;

+ (Braze *)braze {
  return _braze;
}

+ (void)setBraze:(Braze *)braze {
  _braze = braze;
}
```

NotificationService.swift
```swift
import BrazeNotificationService

class NotificationService: BrazeNotificationService.NotificationService {}
```

**Comment by jerielng** [comment]
Thanks for double checking @huextrat! We have a working sample in progress and will get that published in our example app as soon as it's ready. You should be able to compare and contrast those to check for any differences. In the meantime, for your specific case, if you can write into support@braze.com with further details of your project, maybe even including your Podfile setup and corresponding Info.plist files for the appex, that might be a good next step to dive into.

**Comment by huextrat** [comment]
Ok please let me know when you have a branch with rich notification example :)

**Comment by vanessaland** [comment]
Hey @huextrat! We've just [released version 8.3.0](https://github.com/braze-inc/braze-react-native-sdk/releases/tag/8.3.0) of the Braze React Native SDK, which contains [some samples](https://github.com/braze-inc/braze-react-native-sdk/tree/master/BrazeProject/ios) on how to integrate rich notifications.

Let us know if you have any questions! If these are sufficient, feel free to close this issue.

**Comment by jerielng** [comment]
Hey @huextrat! Since we have an end-to-end example for integrating rich push notifications, you should be able to reference that in the sample project and compare it against your own implementation. We'll go ahead and close out this issue, but if you continue to face troubles with media displaying in your app, please write into support@braze.com with some of the code and information you shared above so that our support team can work with you in diagnosing the issue.

Thanks!
